src/core/qgis_sip.h

```diff
@@ -197,6 +197,16 @@
  */
 #define SIP_DOC_TEMPLATE
 
+/*
+ * Sip supports the final keyword since version 4.19.0, earlier than that
+ * we will have build issues because it tries to override final methods.
+ */
+#if SIP_VERSION >= 0x041300
+#define SIP_FINAL final
```

Comment:
just wondering -- could we avoid using SIP_FINAL in the cpp headers and just use "final" instead?

Reply:
And have a macro (in qgis.h?) that redefines final to override for older sip versions?

Reply:
couldn't we do the replacement in sipify?

Reply:
Cpp headers also need to be different.
If you have a `final` in a header and use it with sip 4.18, sip will override this method and the compiler will complain.

Reply:
Ok. I'm late to the discussion :p

Hmm messy either way. I'd prefer a more generic macro if nothing else, even "FINAL" or "QGS_FINAL"

Reply:
I was thinking the same.
And it should actually be defined by qgis.h and redefined by qgis_sip.h for older sip versions. But the effect will be, that someone with an older sip version will have to do an include order cleanup ;P

Reply:
I guess I was able to avoid the todo